### PR TITLE
Listenonrepeat

### LINKED ISSUE #1293: [REQUEST] Add support for https://listenonrepeat.com
Add scrobbling support for the website:
https://listenonrepeat.com

### PR DESCRIPTION
**Pull request by inverse**
Resolves #1293

### CODE REVIEW

File: connectors/listenonrepeat.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+'use strict';
+
+/* global Connector Util */
+
+Connector.playerSelector = '.player-controls';
+
+Connector.artistTrackSelector = '.player-controls > .video-title:first';
+
+Connector.playButtonSelector = '.controls > button:nth-child(4)';
```
Comment (alexesprit) [inline connectors/listenonrepeat.js]: This is redundant, since `isPlaying` function is used.

**Comment by alexesprit** [comment]
I suggest to watch for changes of repeat counter and reset state using `Connector.resetState` function on every change. This approach will allow to support repeating videos.

**Comment by inverse** [comment]
Good call - as you can see it's been a while since I touched this ;) Will update tomorrow or feel free to push to the branch if you wanna get this merged today.

**Comment by inverse** [comment]
@alexesprit what would you use for the selector for the repeat counter? I could use the one generated by chrome but it feels very strict 

```
#page-home > div:nth-child(2) > div:nth-child(3) > div > div:nth-child(3) > div > div:nth-child(4) > div > div:nth-child(1)
```

Also would you attach this with a mutation observer?

**Comment by alexesprit** [comment]
@inverse Looks like the repeat counter is not available when the connector is injected. We should wait until the counter element is visible using another MutationObserver. I'll try to implement something like that later.

> #page-home > div:nth-child(2) > div:nth-child(3) > div > div:nth-child(3) > div > div:nth-child(4) > div > div:nth-child(1)

That's no good. I think we could use `.hide-on-small-only svg + span` or parent of parent of `.hide-on-small-only #Icon` but they're no much better than generated one by Chrome.

**Comment by alexesprit** [comment]
I had no luck with adding of repeat support, so I think the connector works as much as it can.

**Comment by f1u77y** [comment]
@alexesprit 
> repeat support

Couldn't we just inject code into the iframe and use existing YT code that should support repeat?

**Comment by alexesprit** [comment]
@f1u77y AFAIK we are not able to do it due to same-origin security policy.

**Comment by inverse** [comment]
That sucks :( @alexesprit what issue did you have? getting access to the listen counter? 

I wish they had nicer semantics in their DOM.

**Comment by f1u77y** [comment]
@alexesprit we can inject code that will interact with main frame via `window.postMessage()`

**Pull request merged**